corrupted gz in the cache

We use an API key for downloads. When downloading from services.nvd.nist.gov, we sometimes encounter connection problems. After everything seemed to be working fine, our Jenkins build pipeline was able to load the nvdcve-modified.gz from the cache but failed to unpack it. The nvdcve-modified.gz could not be unpacked even manually afterwards. After deleting the file nvdcve-modified.gz and restarting the container, the file was then present correctly. Can a validation for *.gz be built in after creation/download into the cache to ensure that it is not provided in a corrupted state? It would be nice to have retry mechanisms and logging in place.

```
Retrying request /rest/json/cves/2.0?resultsPerPage=2000&startIndex=256000 : 2 time
Recoverable I/O exception (org.apache.hc.core5.http.RequestNotExecutedException) caught when processing request to {s}->https://services.nvd.nist.gov:443

Retrying request /rest/json/cves/2.0?resultsPerPage=2000&startIndex=262000 : 2 time
Recoverable I/O exception (org.apache.hc.core5.http.RequestNotExecutedException) caught when processing request to {s}->https://services.nvd.nist.gov:443

Completed in 9675 seconds
```

Thanks for the nice project.

"We've discovered an OutOfMemoryException in older logs, but since this occurred some time ago, the following is merely speculation. It's conceivable that this exception might have also occurred while writing GZ files to the disk, even if it wasn't captured in the logs. This could potentially explain why a corrupt file was created. However, this doesn't immediately account for why the log output 'Completed in 9675 seconds' was still recorded."

The filesize from the corrupted file was 130200749.
"GET /nvdcve-modified.json.gz HTTP/1.1" 200 130200749

Here is the corresponding log excerpt from the Jenkins pipeline:
`[INFO] Download Started for NVD Cache - https://nvd.optadata.de/nvdcve-modified.json.gz`
 `[ERROR] java.util.concurrent.ExecutionException: com.fasterxml.jackson.databind.JsonMappingException: Unexpected character ('1' (code 49)): was expecting comma to separate Object entries
 at [Source: REDACTED (StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION disabled); line: 1, column: 772236742] (through reference chain: io.github.jeremylong.openvulnerability.client.nvd.DefCveItem["cve"]->io.github.jeremylong.openvulnerability.client.nvd.CveItem["configurations"]->java.util.ArrayList[815]->io.github.jeremylong.openvulnerability.client.nvd.Config["nodes"]->java.util.ArrayList[1]->io.github.jeremylong.openvulnerability.client.nvd.Node["cpeMatch"]->java.util.ArrayList[0])
 org.owasp.dependencycheck.data.update.exception.UpdateException: java.util.concurrent.ExecutionException: com.fasterxml.jackson.databind.JsonMappingException: Unexpected character ('1' (code 49)): was expecting comma to separate Object entries
 at [Source: REDACTED (StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION disabled); line: 1, column: 772236742] (through reference chain: io.github.jeremylong.openvulnerability.client.nvd.DefCveItem["cve"]->io.github.jeremylong.openvulnerability.client.nvd.CveItem["configurations"]->java.util.ArrayList[815]->io.github.jeremylong.openvulnerability.client.nvd.Config["nodes"]->java.util.ArrayList[1]->io.github.jeremylong.openvulnerability.client.nvd.Node["cpeMatch"]->java.util.ArrayList[0])
 at org.owasp.dependencycheck.data.update.NvdApiDataSource.processFuture (NvdApiDataSource.java:265)`

--- a/vulnz/src/docker/scripts/mirror.sh
+++ b/vulnz/src/docker/scripts/mirror.sh
@@ -33,3 +33,11 @@ if [ -n "${DEBUG}" ]; then
 fi
 
 java $JAVA_OPT -jar /usr/local/bin/vulnz cve $DELAY_ARG $DEBUG_ARG $MAX_RETRY_ARG $MAX_RECORDS_PER_PAGE_ARG --cache --directory /usr/local/apache2/htdocs
+
+echo "Valdiating the cache..."
+
+if ! find /usr/local/apache2/htdocs -name "*.gz" -type f -exec gzip -t {} \; ; then
+    echo "Corrupt gz files detected, clearing cache and re-running mirror"
+    rm -rf /usr/local/apache2/htdocs/*
+    java $JAVA_OPT -jar /usr/local/bin/vulnz cve $DELAY_ARG $DEBUG_ARG $MAX_RETRY_ARG $MAX_RECORDS_PER_PAGE_ARG --cache --directory /usr/local/apache2/htdocs
+fi

--- a/vulnz/src/docker/scripts/mirror.sh
+++ b/vulnz/src/docker/scripts/mirror.sh
@@ -34,10 +34,12 @@ fi
 
 java $JAVA_OPT -jar /usr/local/bin/vulnz cve $DELAY_ARG $DEBUG_ARG $MAX_RETRY_ARG $MAX_RECORDS_PER_PAGE_ARG --cache --directory /usr/local/apache2/htdocs
 
-echo "Valdiating the cache..."
-
-if ! find /usr/local/apache2/htdocs -name "*.gz" -type f -exec gzip -t {} \; ; then
-    echo "Corrupt gz files detected, clearing cache and re-running mirror"
-    rm -rf /usr/local/apache2/htdocs/*
-    java $JAVA_OPT -jar /usr/local/bin/vulnz cve $DELAY_ARG $DEBUG_ARG $MAX_RETRY_ARG $MAX_RECORDS_PER_PAGE_ARG --cache --directory /usr/local/apache2/htdocs
-fi
+echo "Validating the cache..."
+for file in /usr/local/apache2/htdocs/*.gz; do
+    if ! gzip -t "$file"; then
+        echo "Corrupt gz file detected: $file, clearing cache and re-running mirror"
+        rm -rf /usr/local/apache2/htdocs/*
+        java $JAVA_OPT -jar /usr/local/bin/vulnz cve $DELAY_ARG $DEBUG_ARG $MAX_RETRY_ARG $MAX_RECORDS_PER_PAGE_ARG --cache --directory /usr/local/apache2/htdocs
+        break
+    fi
+done

